### PR TITLE
Updated Neo4j Integration to add metrics introduced into Neo4j since 5.10

### PR DESCRIPTION
**Pull request by davidfauth**
Updated the conf.yaml.example to change default version to 5.18 and to remove reference to Neo4j 3.5w

### What does this PR do?

Neo4j has introduced new metrics since the last Neo4j integration release which was accurate up to Neo4j 5.10.
Additionally, the conf.yaml.example file had references to Neo4j 3.5 which were removed. I also changed the default Neo4j version and attempted to clarify how to specify databases to monitor.

### Motivation

Additional metrics have been added that should be available to users.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by brett0000FF — APPROVED** [review]

**Review by emarsha94 — APPROVED** [review]

**Review by mgarabed — APPROVED** [review]

File: neo4j/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # CHANGELOG - neo4j
 
+## 3.0.3 / 2024-03-29
+
+***Added***:
+
+* Added additional bolt, page_cache eviction and relationship type  metrics which have been introduced since Neo4j 5.10
+
+***Changed***:
```
Comment (steveny91) [inline neo4j/CHANGELOG.md:9]: Can you move the changed section to above the added section? That's currently failing validation right now.

**Comment by davidfauth** [comment]
Updated CHANGELOG.md as requested.

**Review by steveny91 — APPROVED** [review]

**Comment by davidfauth** [comment]
Can I resolve conflicts or is that something you have to do?

**Comment by mgarabed** [comment]
> Can I resolve conflicts or is that something you have to do?

@davidfauth looks like there was a version bump to `3.0.3` that happened in this PR: https://github.com/DataDog/integrations-extras/pull/2314

Can you bump the version here to `3.0.4` which should then address the conflicts?

**Comment by emarsha94** [comment]
> > Can I resolve conflicts or is that something you have to do?
> 
> @davidfauth looks like there was a version bump to `3.0.3` that happened in this PR: #2314
> 
> Can you bump the version here to `3.0.4` which should then address the conflicts?

@davidfauth bumping this comment!

**Review by mgarabed — APPROVED** [review]

**Pull request merged**